### ISSUE TITLE
Fairness without demographics in repeated loss minimization

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#370]
Fairness without demographics in repeated loss minimization<br>
<br>
https://ift.tt/2Pl30Ge<br>
<br>
adriancolyer<br>
<br>
<div>
<div class="entry clear">
<p><a href="https://arxiv.org/abs/1806.08010">Fairness without demographics in repeated loss minimization</a> Hashimoto et al., <em>ICML&rsquo;18</em></p>
<p>When we train machine learning models and optimise for average loss it is possible to obtain systems with very high overall accuracy, but which perform poorly on under-represented subsets of the input space. For example, a speech recognition system that performs poorly with minority accents.</p>
<blockquote>
<p>We refer to this phenomenon of high overall accuracy but low minority accuracy as a <em>representation disparity</em>&hellip; This representation disparity forms our definition of unfairness, and has been observed in face recognition, language identification, dependency parsing, part-of-speech tagging, academic recommender systems, and automatic video captioning.</p>
</blockquote>
<p>For systems that are continually trained and evolved based on data collected from their users, the poor performance for a minority group can set in place a vicious cycle in which members of such a group use the system less (because it doesn&rsquo;t work as well for them), causing them to provide less data and hence to be further under-represented in the training set&hellip;</p>
<blockquote>
<p>&hellip; this problem of <em>disparity amplification</em> is a possibility in any machine learning system that is retrained on user data.</p>
</blockquote>
<p>An interesting twist in the problem is that the authors assume neither the number of groups (<em>K</em>), nor group membership, is known. An online speech recognition system for example, is unlikely to collect demographic information on its users. We simply assume that there will be groups, and then bound the worst-case risk for any group.</p>
<blockquote>
<p>Our work can be seen as a direct instantiation of <a href="https://en.m.wikipedia.org/wiki/Justice_as_Fairness:_A_Restatement">John Rawl&rsquo;s theory on distributive justice and stability</a>, where we view predictive accuracy as a resource to be allocated. Rawls argues that the <em>difference principle</em>, defined as maximizing the welfare of the worst-off group, is fair and stable over time since it ensures that minorities consent to and attempt to maintain the status quo.</p>
</blockquote>
<p>After showing that representation disparity and disparity amplification are serious issues with the current status quo, the authors go on to introduce a method based on <em>distributionally robust optimization</em> (DRO) which can can control the worst-case risk.</p>
<blockquote>
<p>We demonstrate that DRO provides a upper bound on the risk incurred by minority groups and performs well in practice. Our proposed algorithm is straightforward to implement, and induces distributional robustness, which can be viewed as a benefit in and of itself.</p>
</blockquote>
<h3>Representation disparity and disparity amplification</h3>
<p>We have a population with <img src="https://s0.wp.com/latex.php?latex=K&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="K" title="K" class="latex"> <em>latent</em> groups. Each group <img src="https://s0.wp.com/latex.php?latex=k+%5Cin+K&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="k \in K" title="k \in K" class="latex"> is a proportion <img src="https://s0.wp.com/latex.php?latex=%5Calpha_k&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="\alpha_k" title="\alpha_k" class="latex"> of the overall population, and has a distribution <img src="https://s0.wp.com/latex.php?latex=P_k&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="P_k" title="P_k" class="latex">. We have some trained model with parameters <img src="https://s0.wp.com/latex.php?latex=%5Ctheta&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="\theta" title="\theta" class="latex"> used to make predictions based on an input <img src="https://s0.wp.com/latex.php?latex=Z&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="Z" title="Z" class="latex"> arising from one of the latent groups. When a user makes a query, they incur some expected loss, which is termed the <em>risk</em> <img src="https://s0.wp.com/latex.php?latex=R%28%5Ctheta%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R(\theta)" title="R(\theta)" class="latex"> in this paper.</p>
<p><strong>Representation disparity</strong> occurs when there is a meaningful gap between the overall risk <img src="https://s0.wp.com/latex.php?latex=R%28%5Ctheta%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R(\theta)" title="R(\theta)" class="latex"> and the maximum (worst-case) group risk <img src="https://s0.wp.com/latex.php?latex=R_%7Bk%7D%28%5Ctheta%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R_{k}(\theta)" title="R_{k}(\theta)" class="latex">. I.e., good overall performance, but poor performance (higher expected loss) for the worst-case group in <em>K</em>. In the evaluation example, the task is text prediction for autocomplete and the training data set contains a mix of African-American English (AAE) and Standard-American English (SAE) tweets. If the AAE data is in the minority, we would expect worse than average model performance when autocompleting African-American English. The measure of the representation disparity is given by <img src="https://s0.wp.com/latex.php?latex=R_%7BAAE%7D+-+R_%7BAAE%2BSAE%7D&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R_{AAE} - R_{AAE+SAE}" title="R_{AAE} - R_{AAE+SAE}" class="latex">.</p>
<p><strong>Disparity amplification</strong> is the process by which unbalanced group risk gets worse over time.</p>
<p>Consider a system which learns from its users, with periodic re-training. Say there are <em>T</em> rounds of such training. Within each round we might gain some new users and lose some existing users (aka churn). The authors make the reasonable assumption that user retention within a group is a strictly decreasing function of risk level (i.e., the worse the system performs for a group, the fewer users of that group are retained). The decrease in user retention for a minority group as compared to the average causes the group size to shrink, meaning that in the next round it will receive even higher losses due to having fewer samples from the group in the training data.</p>
<p>To keep a handle on the representation disparity, we want to control (minimize) the worst-case group risk over all time periods <img src="https://s0.wp.com/latex.php?latex=t&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="t" title="t" class="latex"> in <img src="https://s0.wp.com/latex.php?latex=1..T&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="1..T" title="1..T" class="latex">.</p>
<blockquote>
<p>Roughly speaking, minimizing the worst-case risk should mitigate disparity amplification as long as lower losses sead to higher user retention.</p>
</blockquote>
<p>For a system to be fair over the long term there needs to be a stable and fair fixed-point for the expected user counts <img src="https://s0.wp.com/latex.php?latex=%5Clambda%5E%7B%28t%29%7D&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="\lambda^{(t)}" title="\lambda^{(t)}" class="latex"> over time. That is, risk minimisation needs to maintain the same population fraction over time. It&rsquo;s easy to violate this condition when minimising overall risk (the standard <em>empirical risk minimization</em>, ERM, approach), even when we start out with two groups that have <em>identical</em> population size and initial risk.</p>
<p>In the following example we have two initially equal groups (one on the left, one on the right), and a classifier tasked with classifying points as red or black. Start out with a classifier with boundary <img src="https://s0.wp.com/latex.php?latex=x_2+%3D+0&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="x_2 = 0" title="x_2 = 0" class="latex"> (optimal for this data set). Now let optimisation play out over <img src="https://s0.wp.com/latex.php?latex=t+%3D+500&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="t = 500" title="t = 500" class="latex"> rounds. At some point in the earlier rounds, due to random fluctuations in sampling accuracy, we took slightly fewer samples from the cluster on the right, meaning that ERM improves the loss for the left cluster at the expense of the right-hand one (see e.g. the <img src="https://s0.wp.com/latex.php?latex=t+%3D+100&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="t = 100" title="t = 100" class="latex"> line). This increases churn in the right-hand cluster, reducing its representation in future rounds. Once the scales have tipped even slightly like this the disparity is amplified until at <img src="https://s0.wp.com/latex.php?latex=t+%3D+500&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="t = 500" title="t = 500" class="latex"> there are nearly no samples from the right-hand cluster and it ends up suffering high loss.</p>
<p><img src="https://adriancolyer.files.wordpress.com/2018/08/dro-fig-1.jpeg?w=520"></p>
<p>Whenever decreasing the risk for one group increases the risk for others sufficiently the fixed point will be unstable and the model will converge to a different, possibly unfair, fixed point.</p>
<h3>Distributionally robust optimization</h3>
<p>Clearly we might not want to use ERM if we are concerned about representation disparity and its amplification. But what should we use instead? We need something that minimises the worst-case risk over time. If we can control the worst-case risk in one time step, then by an induction-like argument you can see that we can control it over all time steps.</p>
<blockquote>
<p>The fundamental difficulty in controlling the worst-case risk over a single time step comes from not observing the group memberships from which the data was sampled&hellip; To achieve reasonable performance across different groups, we postulate a formulation that protects against <em>all</em> directions around the data generating distribution.</p>
</blockquote>
<p>Distributionally robust optimization, DRO, has roots going back to 1958 (Ye, &lsquo;<a href="https://web.stanford.edu/~yyye/DRSP.pdf">Distributionally robust stochastic and online optimization</a>&rsquo; ) and deals with optimisation in situations where you are not sure of (or an adversary controls, within some bounds), the true underlying distribution.</p>
<p>In this case, what the authors do is consider all possible distributions <img src="https://s0.wp.com/latex.php?latex=Q&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="Q" title="Q" class="latex"> within a distance <em>r</em> of the overall population distribution <img src="https://s0.wp.com/latex.php?latex=P&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="P" title="P" class="latex">, where <em>r</em> is the <a href="https://en.m.wikipedia.org/wiki/F-divergence">chi-squared divergence</a> measure between the two distributions. The result is a &lsquo;<em>chi-squared ball</em>&rsquo;. Let <img src="https://s0.wp.com/latex.php?latex=R_%7Bdro%7D%28%5Ctheta%3Br%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R_{dro}(\theta;r)" title="R_{dro}(\theta;r)" class="latex"> be the worst-case loss over all the r-perturbations around P. The authors show that this bounds the risk of the group <img src="https://s0.wp.com/latex.php?latex=R_%7Bk%7D%28%5Ctheta%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R_{k}(\theta)" title="R_{k}(\theta)" class="latex"> and hence the overall worst-case risk.</p>
<p>If the smallest minority group proportion we want to consider is <img src="https://s0.wp.com/latex.php?latex=%5Calpha_%7Bmin%7D&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="\alpha_{min}" title="\alpha_{min}" class="latex">, then we can control the worst-case group risk by minimizing the upper bound of <img src="https://s0.wp.com/latex.php?latex=R_%7Bdro%7D%28%5Ctheta%3Br_%7Bmax%7D%29&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="R_{dro}(\theta;r_{max})" title="R_{dro}(\theta;r_{max})" class="latex"> with <img src="https://s0.wp.com/latex.php?latex=r_%7Bmax%7D+%3A%3D+%281%2F%5Calpha_%7Bmin%7D+-1%29%5E2&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="r_{max} := (1/\alpha_{min} -1)^2" title="r_{max} := (1/\alpha_{min} -1)^2" class="latex">. See &sect;4.1 in the paper for details. Under the covers what happens is that small levels of loss get ignored, and large losses above a threshold are upweighted due to a square term. My statistics knowledge isn&rsquo;t good enough to take you any deeper here!</p>
<p>Fortunately, if you just want to put this into practice without diving into all the underlying theory, you just need to need to use the following optimisation objective:</p>
<p><img src="https://adriancolyer.files.wordpress.com/2018/08/dro-eqn-6.jpeg?w=480"></p>
<p>Where <img src="https://s0.wp.com/latex.php?latex=%5Ceta&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="\eta" title="\eta" class="latex"> is a hyperparameter.</p>
<h3>Evaluation</h3>
<p>Recall the experiment with two equal left and right groups and a red/black classifier that we looked at earlier. This was unstable under ERM, with a minority group emerging and classification accuracy for that minority group getting worse over time. If we run the same experiment with DRO, the disparity amplification effect disappears.</p>
<p><img src="https://adriancolyer.files.wordpress.com/2018/08/dro-fig-5.jpeg?w=480"></p>
<p>The authors also complete a real-world human evaluation of user retention and satisfaction on a text autocomplete task (with a mix of African-American Engish and Standard-American English training examples as described earlier).</p>
<blockquote>
<p>For both ERM and DRO, we train a set of five maximum likelihood bigram language models on a corpus with 366,361 tweets total and a <img src="https://s0.wp.com/latex.php?latex=f+%5Cin+%7B0.1%2C+0.4%2C+0.5%2C+0.6%2C+0.9%7D&amp;bg=ffffff&amp;fg=333333&amp;s=0" alt="f \in {0.1, 0.4, 0.5, 0.6, 0.9}" title="f \in {0.1, 0.4, 0.5, 0.6, 0.9}" class="latex"> fraction of the tweets labelled as AAE. This results in 10 possible autocomplete systems a given Mechanical Turk user can be assigned to during a task.</p>
</blockquote>
<p>The results show an improvement in both minority satisfaction and retention rate with DRO, with only slightly decreasing scores for the SAE group.</p>
<p><img src="https://adriancolyer.files.wordpress.com/2018/08/dro-fig-4.jpeg?w=640"><br>
(<a href="https://adriancolyer.files.wordpress.com/2018/08/dro-fig-4.jpeg">Enlarge</a>)</p>
<p>The code to generate results is <a href="https://worksheets.codalab.org/worksheets/0x17a501d37bbe49279b0c70ae10813f4c/">available on CodaLab</a>. (<a href="https://worksheets.codalab.org/">CodaLab</a> itself looks pretty interesting btw. &ndash; &lsquo;a collaborative platform for reproducible research&rsquo;).</p>
</div>
</div>
<br>
<br>
<br>
<br>
BigData,AI|ML|DM|IR|PR<br>
<br>
via the morning paper https://ift.tt/1WJ1E4W<br>
<br>
August 17, 2018 at 01:00PM
